### ISSUE TITLE
Trouble to get OpenFaas pods running on Raspberry Pi 3A+ : Gateway and Queue-Worker pods keep crashing

### ISSUE THREAD

**Issue by ZeusIV** [openfaas/faas-netes#332]
Hello,
I got some trouble with the deployment of OpenFaas on my RaspPi 3B+ 3A+ K8s cluster.
Two pods, gateway and queue-worker are in a CrashLoopBackOff state.

This is pretty much a duplicate of the closed issue https://github.com/openfaas/faas-netes/issues/232
I first read about this over at https://www.hanselman.com/blog/HowToBuildAKubernetesClusterWithARMRaspberryPiThenRunNETCoreOnOpenFaas.aspx
and came across some posts about rpi Zero clusters. By the time I got around to giving it a go several things no longer worked with arm6 which  was a bit of a setback. I didn't want to splash out on several rpi 3B+. Then along came the rpi 3A+ which is the same as the rpi 3B+ without all of the io. So I built a cluster with 1 x rpi 3B+ and 4 x rpi 3A+. Followed the instructions at https://gist.github.com/alexellis/fdbc90de7691a1b9edb545c17da2d975 and got the markdown example working ok. Took me a couple of hours. Then I got to faas https://docs.openfaas.com/deployment/kubernetes/

$ git clone https://github.com/openfaas/faas-netes
$ kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
$ cd faas-netes && \
kubectl apply -f ./yaml_armhf

## Expected Behaviour
All Openfaas pods should be running.
## Current Behaviour
Two pods are crashing :
### Gateway pod
```
$ kubectl logs gateway-cdbdbcd54-xrvxp -n openfaas -c gateway
2018/12/07 14:47:36 HTTP Read Timeout: 1m0s
2018/12/07 14:47:36 HTTP Write Timeout: 1m0s
2018/12/07 14:47:36 Binding to external function provider: http://127.0.0.1:8081/
2018/12/07 14:47:36 Async enabled: Using NATS Streaming.
2018/12/07 14:47:36 Opening connection to nats://nats.openfaas:4222
2018/12/07 14:47:38 nats: no servers available for connection
```
### Queue-Worker pod

```
$ kubectl logs queue-worker-86df6f8747-cbrzf -n openfaas
Can't connect: nats: no servers available for connection

```

### Nats pod
```

$ kubectl logs nats-765f98bd59-wcx68 -n openfaas
[1] 2018/12/07 14:20:14.881527 [INF] STREAM: Starting nats-streaming-server[faas-cluster] version 0.11.2
[1] 2018/12/07 14:20:14.881774 [INF] STREAM: ServerID: UkFQh0EO149uRUWzbwseqv
[1] 2018/12/07 14:20:14.881812 [INF] STREAM: Go version: go1.11.1
[1] 2018/12/07 14:20:14.884280 [INF] Starting nats-server version 1.3.0
[1] 2018/12/07 14:20:14.884385 [INF] Git commit [not set]
[1] 2018/12/07 14:20:14.889054 [INF] Listening for client connections on 0.0.0.0:4222
[1] 2018/12/07 14:20:14.889118 [INF] Server is ready
[1] 2018/12/07 14:20:14.984437 [INF] STREAM: Recovering the state...
[1] 2018/12/07 14:20:14.984524 [INF] STREAM: No recovered state
[1] 2018/12/07 14:20:15.235734 [INF] STREAM: Message store is MEMORY
[1] 2018/12/07 14:20:15.235901 [INF] STREAM: ---------- Store Limits ----------
[1] 2018/12/07 14:20:15.235919 [INF] STREAM: Channels:                  100 *
[1] 2018/12/07 14:20:15.235933 [INF] STREAM: --------- Channels Limits --------
[1] 2018/12/07 14:20:15.235946 [INF] STREAM:   Subscriptions:          1000 *
[1] 2018/12/07 14:20:15.235959 [INF] STREAM:   Messages     :       1000000 *
[1] 2018/12/07 14:20:15.235971 [INF] STREAM:   Bytes        :     976.56 MB *
[1] 2018/12/07 14:20:15.235984 [INF] STREAM:   Age          :     unlimited *
[1] 2018/12/07 14:20:15.235997 [INF] STREAM:   Inactivity   :     unlimited *
[1] 2018/12/07 14:20:15.236009 [INF] STREAM: ----------------------------------
```
### events
```
 $ kubectl get events --sort-by=.metadata.creationTimestamp -n openfaas
LAST SEEN   TYPE      REASON              KIND         MESSAGE
50m         Normal    ScalingReplicaSet   Deployment   Scaled up replica set alertmanager-c6c9696c4 to 1
50m         Normal    Scheduled           Pod          Successfully assigned openfaas/prometheus-84cfb5cdc7-rch2q to cube3
2s          Warning   BackOff             Pod          Back-off restarting failed container
50m         Normal    SuccessfulCreate    ReplicaSet   Created pod: prometheus-84cfb5cdc7-rch2q
50m         Warning   FailedCreate        ReplicaSet   Error creating: pods "gateway-cdbdbcd54-" is forbidden: error looking up service account openfaas/faas-controller: serviceaccount "faas-controller" not found
50m         Normal    SuccessfulCreate    ReplicaSet   Created pod: alertmanager-c6c9696c4-m4p6q
50m         Warning   FailedMount         Pod          MountVolume.SetUp failed for volume "prometheus-config" : configmap "prometheus-config" not found
50m         Normal    Scheduled           Pod          Successfully assigned openfaas/queue-worker-86df6f8747-cbrzf to cube4
50m         Normal    ScalingReplicaSet   Deployment   Scaled up replica set prometheus-84cfb5cdc7 to 1
50m         Normal    SuccessfulCreate    ReplicaSet   Created pod: nats-765f98bd59-wcx68
50m         Normal    Scheduled           Pod          Successfully assigned openfaas/nats-765f98bd59-wcx68 to cube2
50m         Normal    ScalingReplicaSet   Deployment   Scaled up replica set gateway-cdbdbcd54 to 1
50m         Normal    Pulling             Pod          pulling image "nats-streaming:0.11.2"
48m         Normal    Started             Pod          Started container
50m         Normal    Pulled              Pod          Successfully pulled image "nats-streaming:0.11.2"
50m         Normal    Scheduled           Pod          Successfully assigned openfaas/alertmanager-c6c9696c4-m4p6q to cube2
50m         Normal    Pulling             Pod          pulling image "openfaas/faas-netes:0.6.3-armhf"
48m         Normal    Created             Pod          Created container
50m         Normal    SuccessfulCreate    ReplicaSet   Created pod: gateway-cdbdbcd54-xrvxp
48m         Normal    Pulling             Pod          pulling image "openfaas/gateway:0.9.10-armhf"
50m         Normal    Created             Pod          Created container
50m         Normal    Started             Pod          Started container
48m         Normal    Pulled              Pod          Successfully pulled image "openfaas/gateway:0.9.10-armhf"
50m         Normal    Started             Pod          Started container
50m         Normal    Pulled              Pod          Successfully pulled image "functions/prometheus:2.2.0-armhf"
50m         Normal    Pulling             Pod          pulling image "functions/prometheus:2.2.0-armhf"
50m         Normal    Scheduled           Pod          Successfully assigned openfaas/gateway-cdbdbcd54-xrvxp to cube2
50m         Normal    Pulled              Pod          Successfully pulled image "functions/alertmanager:0.15.0-rc.0-armhf"
50m         Normal    Created             Pod          Created container
50m         Normal    Started             Pod          Started container
50m         Normal    Created             Pod          Created container
48m         Normal    Pulling             Pod          pulling image "openfaas/queue-worker:0.4.9-armhf"
50m         Normal    Pulling             Pod          pulling image "functions/alertmanager:0.15.0-rc.0-armhf"
49m         Normal    Created             Pod          Created container
50m         Normal    ScalingReplicaSet   Deployment   Scaled up replica set queue-worker-86df6f8747 to 1
48m         Normal    Pulled              Pod          Successfully pulled image "openfaas/queue-worker:0.4.9-armhf"
49m         Normal    Started             Pod          Started container
46s         Warning   BackOff             Pod          Back-off restarting failed container
50m         Normal    SuccessfulCreate    ReplicaSet   Created pod: queue-worker-86df6f8747-cbrzf
50m         Normal    ScalingReplicaSet   Deployment   Scaled up replica set nats-765f98bd59 to 1
50m         Normal    Pulled              Pod          Successfully pulled image "openfaas/faas-netes:0.6.3-armhf"
49m         Normal    Created             Pod          Created container
49m         Normal    Started             Pod          Started container
```
### Gateway pod
```
kubectl describe pod gateway-cdbdbcd54-xrvxp -n openfaas
Name:               gateway-cdbdbcd54-xrvxp
Namespace:          openfaas
Priority:           0
PriorityClassName:  <none>
Node:               cube2/192.168.1.22
Start Time:         Fri, 07 Dec 2018 14:19:25 +0000
Labels:             app=gateway
                    pod-template-hash=cdbdbcd54
Annotations:        <none>
Status:             Running
IP:                 10.44.0.2
Controlled By:      ReplicaSet/gateway-cdbdbcd54
Containers:
  gateway:
    Container ID:   docker://d5832ea97ecd9a0f1ebd277fc6202f85275e0fc82a77be44ae9ee777abdffda2
    Image:          openfaas/gateway:0.9.10-armhf
    Image ID:       docker-pullable://openfaas/gateway@sha256:15776190818c6464561eb9c4193c1effeeea3110d56742ceb8a3b1756bfe65a4
    Port:           8080/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Fri, 07 Dec 2018 14:52:46 +0000
      Finished:     Fri, 07 Dec 2018 14:52:48 +0000
    Ready:          False
    Restart Count:  11
    Limits:
      memory:  100Mi
    Requests:
      memory:  100Mi
    Environment:
      functions_provider_url:   http://127.0.0.1:8081/
      faas_nats_address:        nats.openfaas
      faas_nats_port:           4222
      direct_functions:         true
      direct_functions_suffix:  openfaas-fn.svc.cluster.local.
      read_timeout:             60s
      write_timeout:            60s
      basic_auth:               false
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from faas-controller-token-wrqlr (ro)
  faas-netesd:
    Container ID:   docker://bcca62be28b1332e2b34f4ee91385b8ba0f9935f9e2d26a38cf934c222797206
    Image:          openfaas/faas-netes:0.6.3-armhf
    Image ID:       docker-pullable://openfaas/faas-netes@sha256:df563929222e24ff131ce51e9edc75f95cfc135886704eb4f6e0d391944cddd5
    Port:           8081/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Fri, 07 Dec 2018 14:20:19 +0000
    Ready:          True
    Restart Count:  0
    Limits:
      memory:  50Mi
    Requests:
      memory:  50Mi
    Environment:
      port:                8081
      function_namespace:  openfaas-fn
      read_timeout:        50s
      write_timeout:       60s
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from faas-controller-token-wrqlr (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  faas-controller-token-wrqlr:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  faas-controller-token-wrqlr
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  37m                   default-scheduler  Successfully assigned openfaas/gateway-cdbdbcd54-xrvxp to cube2
  Normal   Pulling    36m                   kubelet, cube2     pulling image "openfaas/faas-netes:0.6.3-armhf"
  Normal   Pulled     36m                   kubelet, cube2     Successfully pulled image "openfaas/faas-netes:0.6.3-armhf"
  Normal   Created    36m                   kubelet, cube2     Created container
  Normal   Started    36m                   kubelet, cube2     Started container
  Normal   Pulling    35m (x4 over 37m)     kubelet, cube2     pulling image "openfaas/gateway:0.9.10-armhf"
  Normal   Pulled     35m (x4 over 36m)     kubelet, cube2     Successfully pulled image "openfaas/gateway:0.9.10-armhf"
  Normal   Created    35m (x4 over 36m)     kubelet, cube2     Created container
  Normal   Started    35m (x4 over 36m)     kubelet, cube2     Started container
  Warning  BackOff    118s (x155 over 36m)  kubelet, cube2     Back-off restarting failed container
```

### Queue-worker pod
```
kubectl describe pod queue-worker-86df6f8747-cbrzf -n openfaas
Name:               queue-worker-86df6f8747-cbrzf
Namespace:          openfaas
Priority:           0
PriorityClassName:  <none>
Node:               cube4/192.168.1.24
Start Time:         Fri, 07 Dec 2018 14:19:24 +0000
Labels:             app=queue-worker
                    pod-template-hash=86df6f8747
Annotations:        <none>
Status:             Running
IP:                 10.44.0.1
Controlled By:      ReplicaSet/queue-worker-86df6f8747
Containers:
  queue-worker:
    Container ID:   docker://b8b1661d15846e7362deafe17f720834237a859d4e58616158520e7eb319484b
    Image:          openfaas/queue-worker:0.4.9-armhf
    Image ID:       docker-pullable://openfaas/queue-worker@sha256:60d0e65ad6f2f31dfd60248dd2fc96aea245a703e84dc96edbece9ea4acf55a8
    Port:           <none>
    Host Port:      <none>
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Fri, 07 Dec 2018 14:56:56 +0000
      Finished:     Fri, 07 Dec 2018 14:56:58 +0000
    Ready:          False
    Restart Count:  12
    Environment:
      max_inflight:          1
      ack_wait:              30s
      faas_function_suffix:  .openfaas-fn
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-lk4m8 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  default-token-lk4m8:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-lk4m8
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                  From               Message
  ----     ------     ----                 ----               -------
  Normal   Scheduled  40m                  default-scheduler  Successfully assigned openfaas/queue-worker-86df6f8747-cbrzf to cube4
  Normal   Created    39m (x4 over 40m)    kubelet, cube4     Created container
  Normal   Started    39m (x4 over 40m)    kubelet, cube4     Started container
  Normal   Pulling    38m (x5 over 40m)    kubelet, cube4     pulling image "openfaas/queue-worker:0.4.9-armhf"
  Normal   Pulled     38m (x5 over 40m)    kubelet, cube4     Successfully pulled image "openfaas/queue-worker:0.4.9-armhf"
  Warning  BackOff    13s (x181 over 40m)  kubelet, cube4     Back-off restarting failed container
```
### Nats pod
```
 kubectl describe pod nats-765f98bd59-wcx68 -n openfaas
Name:               nats-765f98bd59-wcx68
Namespace:          openfaas
Priority:           0
PriorityClassName:  <none>
Node:               cube2/192.168.1.22
Start Time:         Fri, 07 Dec 2018 14:19:24 +0000
Labels:             app=nats
                    pod-template-hash=765f98bd59
Annotations:        <none>
Status:             Running
IP:                 10.44.0.3
Controlled By:      ReplicaSet/nats-765f98bd59
Containers:
  nats:
    Container ID:  docker://46fa203ed7bb96fda795829f01fa14a7b1cb770ec313ad5a07fa061de6782f62
    Image:         nats-streaming:0.11.2
    Image ID:      docker-pullable://nats-streaming@sha256:d70706ce2e92efa6daa193731cd3aec98aedf3c15c51c2cf5b7a6bbc80a43044
    Port:          4222/TCP
    Host Port:     0/TCP
    Command:
      /nats-streaming-server
    Args:
      --store
      memory
      --cluster_id
      faas-cluster
    State:          Running
      Started:      Fri, 07 Dec 2018 14:20:12 +0000
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-lk4m8 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  default-token-lk4m8:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-lk4m8
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  41m   default-scheduler  Successfully assigned openfaas/nats-765f98bd59-wcx68 to cube2
  Normal  Pulling    41m   kubelet, cube2     pulling image "nats-streaming:0.11.2"
  Normal  Pulled     41m   kubelet, cube2     Successfully pulled image "nats-streaming:0.11.2"
  Normal  Created    41m   kubelet, cube2     Created container
  Normal  Started    40m   kubelet, cube2     Started container
```

## Your Environment

* FaaS-CLI version ( Full output from: `faas-cli version` ):
```
 faas-cli version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  5b36c50280d961d3aa248bd17f901f4ef2774447
 version: 0.8.1
```
* Docker version `docker version` (e.g. Docker 17.0.05 ):
$ docker version
Client:
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        4d60db4
 Built:             Wed Nov  7 00:57:21 2018
 OS/Arch:           linux/arm
 Experimental:      false
Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:17:57 2018
  OS/Arch:          linux/arm
  Experimental:     false

* What version and distriubtion of Kubernetes are you using? `kubectl version`
kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T21:04:45Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/arm"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T20:56:12Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/arm"}

* Operating System and version (e.g. Linux, Windows, MacOS):
$ cat /etc/*-release
PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
$ cat /etc/*-release
PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

* What network driver are you using and what CIDR? i.e. Weave net / Flannel
Weave Net

* I tried as suggested 

```
kubectl scale -n openfaas deploy gateway --replicas=0
sleep 30
kubectl scale -n openfaas deploy gateway --replicas=1
```
on the nats, gateway, queue-worker to avoid timing issues but still end up with
```
$ kubectl -n openfaas get po,svc
NAME                                READY   STATUS             RESTARTS   AGE
pod/alertmanager-c6c9696c4-m4p6q    1/1     Running            0          54m
pod/gateway-cdbdbcd54-xrvxp         1/2     CrashLoopBackOff   15         54m
pod/nats-765f98bd59-wcx68           1/1     Running            0          54m
pod/prometheus-84cfb5cdc7-rch2q     1/1     Running            0          54m
pod/queue-worker-86df6f8747-cbrzf   0/1     CrashLoopBackOff   15         54m

NAME                   TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)          AGE
service/alertmanager   ClusterIP   10.107.107.236   <none>        9093/TCP         54m
service/faas-netes     ClusterIP   10.99.123.129    <none>        8081/TCP         54m
service/gateway        NodePort    10.98.46.117     <none>        8080:31112/TCP   54m
service/nats           ClusterIP   10.101.45.172    <none>        4222/TCP         54m
service/prometheus     NodePort    10.111.130.128   <none>        9090:31119/TCP   54m
```

**Comment by ZeusIV** [comment]
Further info
After rebooting the cluster the Dashboard pods panel shows
gateway-cdbdbcd54-xrvxp 
openfaas 
cube2 
Waiting: CrashLoopBackOff 
1347
4 days 
subject
more_vert
Back-off restarting failed container 
Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "8b25afeceaf6cf2868081b35766ea9e4186758a89edaa60eb09decf02ad19c14" network for pod "gateway-cdbdbcd54-xrvxp": NetworkPlugin cni failed to set up pod "gateway-cdbdbcd54-xrvxp_openfaas" network: unable to allocate IP address: Post http://127.0.0.1:6784/ip/8b25afeceaf6cf2868081b35766ea9e4186758a89edaa60eb09decf02ad19c14: dial tcp 127.0.0.1:6784: connect: connection refused 
error 
queue-worker-86df6f8747-cbrzf 
openfaas 
cube4 
Terminated: Error 
1347
4 days 
subject
more_vert
Back-off restarting failed container 
Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "a8b5242eb8920eda4c18b265c2171863f1bb8243d3a89ead915be6fe9c9b5dd0" network for pod "queue-worker-86df6f8747-cbrzf": NetworkPlugin cni failed to set up pod "queue-worker-86df6f8747-cbrzf_openfaas" network: unable to allocate IP address: Post http://127.0.0.1:6784/ip/a8b5242eb8920eda4c18b265c2171863f1bb8243d3a89ead915be6fe9c9b5dd0: dial tcp 127.0.0.1:6784: connect: connection refused 

although  I haven't seen this in any of the logs. Is this part of the problem or just an artifact of the node startup sequence?

TIA

**Comment by alexellis** [comment]
Raspberry Pi is less than ideal to run Kubernetes. It sounds like your network config may be wrong? Did you use DHCP or static IPs? Wired or wireless?

**Comment by ivanayov** [comment]
cc @rdimitrov

**Comment by ZeusIV** [comment]
@alexellis Hi alex, I am using static IPs over wifi. the pi 3 A+ doesn't have ethernet. k8s works fine on my NUC development machine but I would like to use a rpi cluster (if it is possible) for a project I am working on.
I did a search for the sandbox error and found this related to flannel although  I am using weave
https://stackoverflow.com/questions/51169728/failed-create-pod-sandbox-rpc-error-code-unknown-desc-networkplugin-cni-fa
Ensure that /etc/cni/net.d and its /opt/cni/bin friend both exist and are correctly populated with the CNI configuration files and binaries on all Nodes. For flannel specifically, one might make use of the flannel cni repo
 ls /etc/cni/net.d/
10-weave.conflist
ls /opt/cni/bin
bridge  flannel     ipvlan    macvlan  ptp     tuning  weave-ipam  weave-plugin-2.5.0
dhcp    host-local  loopback  portmap  sample  vlan    weave-net

I installed weave as per instructions in the gist and checked those files exist on each node although I am not sure they are configured correctly as I thought the installation did that.
cat /etc/cni/net.d/10-weave.conflist
{
    "cniVersion": "0.3.0",
    "name": "weave",
    "plugins": [
        {
            "name": "weave",
            "type": "weave-net",
            "hairpinMode": true
        },
        {
            "type": "portmap",
            "capabilities": {"portMappings": true},
            "snat": true
        }
    ]
}
I am a bit new to this and don't really know where to look next. The dashboard is working fine and that looks like it gets into all corners of k8s so if there was an obvious network problem I would expect that not to work hardly at all maybe.
